opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1016,12 +1014,15 @@ namespace Opm {
                 const unsigned cellIdx = elemCtx.globalSpaceIndex(/*spaceIdx=*/0, /*timeIdx=*/0);
                 const auto& intQuants = elemCtx.intensiveQuantities(/*spaceIdx=*/0, /*timeIdx=*/0);
                 const auto& fs = intQuants.fluidState();
+                const double pv =
+                    ebosSimulator_.model().dofTotalVolume(cellIdx)
+                    * ebosSimulator_.problem().porosity(cellIdx);
```

Comment:
> And in computing the reference pore volume there are lots of nitty-gritty details, that are not reflected in the apparent formula used here (volume times porosity)

these nitty-gritty (NTG, MULTPV, MULTREGP, PORV, etc.) details are actually considered by the way the porosity is computed by EclProblem: https://github.com/OPM/ewoms/blob/master/ebos/eclproblem.hh#L982

As noted in OPM/ewoms#130, this is not very transparent and depends on lot of side conditions (mainly the ECL specification is to blame here, IMO) and on some details of the grid (e.g. how it implements MINPV).

>  I assume it is baked into the porosity.

exactly. I'm quick to admit that this is not exactly pretty, though.

> To sum up, I really don't like the explicit use of a formula here to compute pore volume, where before we used an already-digested number.

the reason why ebos uses porosity instead of pore volume is that the models in eWoms are not specific to the finite volume discretization and concept of a "pore volume" does not make any sense for discretizations that require the problem to be defined in terms of intensive quantities (e.g., finite element methods).